paymentsheet/src/main/java/com/stripe/android/paymentsheet/ExternalPaymentMethodResultHandler.kt

```diff
@@ -0,0 +1,126 @@
+package com.stripe.android.paymentsheet
+
+import android.content.Context
+import android.content.Intent
+import android.os.Parcelable
+import androidx.annotation.RestrictTo
+import com.stripe.android.core.exception.LocalStripeException
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.payments.paymentlauncher.PaymentResult
+import kotlinx.parcelize.Parcelize
+import java.lang.IllegalStateException
+
+/**
+ * Handler used to respond to external payment method confirm results.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object ExternalPaymentMethodResultHandler {
+
+    /**
+     * Updates the PaymentSheet UI to reflect the result of confirming an external payment method.
+     *
+     * Should be called when [ExternalPaymentMethodConfirmHandler.confirmExternalPaymentMethod] completes.
+     */
+    fun onExternalPaymentMethodResult(context: Context, externalPaymentMethodResult: ExternalPaymentMethodResult) {
+        fun updatePaymentSheetWithResult() {
+            context.startActivity(
+                Intent().setClass(context, PaymentSheetActivity::class.java)
+                    // Needed so that we can start the activity even if the context provided is an application context.
+                    .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                    // Needed so that we can return to PaymentSheet even if a merchant external payment activity hasn't
+                    // yet called finish()
+                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                    // Needed so that PaymentSheetActivity#onNewIntent is called
+                    .addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+                    .putExtra(EXTRA_EXTERNAL_PAYMENT_METHOD_RESULT, externalPaymentMethodResult)
+            )
+        }
+
+        fun updateFlowControllerWithResult() {
+            val flowControllerInstance = PaymentSheet.FlowController.instance
```

Comment:
This is tricky. I don't think we can use the context to get a reference to the flow controller.

I'm also wondering what happens after process death, and this is called before the merchant re instantiates flow controller.

I'm wondering if maybe our best option here is to launch (for result) an invisible activity when the merchant calls flowController.confirm. Then we relaunch that activity here, more similar to how we handle payment sheet.

Maybe we could do this idea for both payment sheet and flow controller to make things consistent and less error prone.

Reply:
Updated the PR to use the invisible activity idea. It's much better now I think! Thanks for the great idea